sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py

```diff
@@ -493,18 +493,341 @@ def request_id(self):
+        """
+        return self._ekus
+
+    @property
+    def key_usage(self):
+        # type: () -> list[KeyUsageType]
+        """List of key usages.
+
+        :rtype: list[~azure.keyvault.certificates.enums.KeyUsageType]
+        """
+        return self._key_usage
+
+    @property
+    def validity_in_months(self):
+        # type: () -> int
+        """The duration that the certificate is valid for in months.
+
+        :rtype: int
+        """
+        return self._validity_in_months
+
+    @property
+    def san_emails(self):
+        # type: () -> list[str]
+        """The subject alternative email addresses.
+
+        :rtype: list[str]
+        """
+        return self._san_emails
+
+    @property
+    def san_dns_names(self):
+        # type: () -> list[str]
+        """The subject alternative domain names.
+
+        :rtype: list[str]
+        """
+        return self._san_dns_names
+
+    @property
+    def san_upns(self):
+        # type: () -> list[str]
+        """The subject alternative user principal names.
+
+        :rtype: list[str]
+        """
+        return self._san_upns
+
+
+class IssuerParameters(object):
+    """Parameters for the issuer of the X509 component of a certificate.
+
+    :param issuer_name: Name of the referenced issuer object or reserved names; for
+     example, 'Self' or 'Unknown'.
+    :type name: str
+    :param certificate_type: Type of certificate to be requested from the
+     issuer provider.
+    :type certificate_type: str
+    :param certificate_transparency: Indicates if the certificates generated
+     under this policy should be published to certificate transparency logs.
+    :type certificate_transparency: bool
+    """
+
+    def __init__(
+        self,
+        issuer_name,  # type: str
+        certificate_type=None,  # type: str
+        certificate_transparency=None  # type: bool
+    ):
+        self._issuer_name = issuer_name
+        self._certificate_type = certificate_type
+        self._certificate_transparency = certificate_transparency
+
+    @classmethod
+    def _from_issuer_parameters_bundle(cls, issuer_parameters_bundle):
+        # type: (models.IssuerParameters) -> IssuerParameters
+        """Converts an autorest-generated IssuerParameters to an IssuerParameters"""
+        return cls(
+            issuer_name=issuer_parameters_bundle.name,
+            certificate_type=issuer_parameters_bundle.certificate_type,
+            certificate_transparency=issuer_parameters_bundle.certificate_transparency
+        )
+
+    def _to_issuer_parameters_bundle(self):
+        # type: (IssuerParameters) -> models.IssuerParameters
+        """Converts an IssuerParameters to an autorest-generated IssuerParameters"""
+        return models.IssuerParameters(
+            name=self.issuer_name,
+            certificate_type=self.certificate_type,
+            certificate_transparency=self.certificate_transparency
+        )
+
+    @property
+    def issuer_name(self):
```

Comment:
I agree this would be better as `name`.

Reply:
we're not longer adding back X509Properties so I'll keep it as issuer_name top level for clarity and to avoid conflicts. Thanks!